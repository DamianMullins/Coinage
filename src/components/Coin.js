import React from 'react';
import PropTypes from 'prop-types';

import { CoinListItem } from '../styles';
import CoinContext from '../context/coinContext';

const Coin = ({ coin }) => (
  <CoinListItem>
    <h3>{coin.name}</h3>

    <CoinContext.Consumer>
      {handleOwnedChange => (
        <label data-testid="coin-label">
          <input
            type="checkbox"
            checked={coin.ownedId !== undefined}
            onChange={handleOwnedChange}
            value={coin.id}
          />
          I own this
        </label>
      )}
    </CoinContext.Consumer>
  </CoinListItem>
);

Coin.propTypes = {
  coin: PropTypes.object.isRequired
};

export default Coin;
