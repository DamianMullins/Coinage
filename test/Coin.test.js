import React from 'react';
import { render, Simulate, flushPromises } from 'react-testing-library';
import Coin from '../src/components/Coin';

it('coin markup is correct', () => {
  // Arrange
  const coin = { id: 2, name: 'B - Bond... James Bond' };

  // Act
  const { container } = render(
    <Coin coin={coin} handleOwnedCheckboxChange={() => {}} />
  );

  // Assert
  expect(container.firstChild).toMatchSnapshot();
});

it('label text content is correct', () => {
  // Arrange
  const coin = { id: 2, name: 'B - Bond... James Bond' };

  // Act
  const { getByTestId } = render(
    <Coin coin={coin}/>
  );

  // Assert
  expect(getByTestId('coin-label').textContent).toBe('I own this');
});
