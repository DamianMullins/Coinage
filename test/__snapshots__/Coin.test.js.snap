// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`coin markup is correct 1`] = `
<label
  data-testid="coin-label"
>
  <input
    type="checkbox"
    value="2"
  />
  B - Bond... James Bond
  <br />
</label>
`;
